src/libraries/System.Text.Json/src/System/Text/Json/AppContextSwitchHelper.cs

```diff
@@ -5,9 +5,13 @@ namespace System.Text.Json
 {
     internal static class AppContextSwitchHelper
     {
-        public static bool IsSourceGenReflectionFallbackEnabled => s_isSourceGenReflectionFallbackEnabled;
+        public static bool IsDefaultReflectionDisabled { get; } =
+            AppContext.TryGetSwitch(
+                switchName: "System.Text.Json.Serialization.DisableDefaultReflection",
+                isEnabled: out bool value)
+            ? value : false;
 
-        private static readonly bool s_isSourceGenReflectionFallbackEnabled =
+        public static bool IsSourceGenReflectionFallbackEnabled { get; } =
             AppContext.TryGetSwitch(
                 switchName: "System.Text.Json.Serialization.EnableSourceGenReflectionFallback",
```

Comment:
I had no idea this option existed 🤔
Just double checking, this is disabled by default, right? And is there any other related MSBuild property that the generator reads that one has to set in order for it to know not to generate those reflection fallback backs? Essentially what I'm wondering is: if one only wants to use generated metadata and no reflection at all, is there anything else other than setting the new "System.Text.Json.Serialization.DisableDefaultReflection" `AppContext` switch that they should know about and do? 😄

Reply:
Correct, it's for enabling .NET 6 behavior that was removed in .NET 7:

https://learn.microsoft.com/en-gb/dotnet/core/compatibility/serialization/7.0/reflection-fallback#use-an-appcontext-switch